traitsui/editors/list_editor.py

```diff
@@ -84,6 +86,18 @@ class ListEditor(EditorFactory):
     #: Show a right-click context menu for the notebook tabs?  (Qt only)
     show_notebook_menu = Bool(False)
 
+    #: Factory callable that will be called to create the new element to add to
+    #: this list. If None, the default value for the trait of interest is used.
+    default_factory = Callable()
```

Comment:
i'd call this an `item_factory` because that's what it is - a factory function to create items - in the context of the list trait. `list_item_factory` felt unnecessarily verbose.